CONTRIBUTING.md

```diff
@@ -1,36 +1,60 @@
-1. Fork the repository.
-2. Clone the forked repository to your local machine.
-3. Follow [the README](README.md) to setup your local repository.
-4. Create a new branch for the issue you're working on; the branch name should start with the issue number and by concise but descriptive, e.g. `123-login-with-locale-crash`.
-5. Make changes to the codebase.
-6. Push your changes to your forked repository.
+1. Fork the repository
+2. Clone the forked repository to your local machine
+3. Follow [the README](README.md) to setup your dev environment
+4. Create a new branch for the issue you're working on; the branch name should start with the issue number and be concise but descriptive, e.g. `123-login-with-locale-crash`
+5. Make changes to the code
+6. Push your changes to your forked repository
+7. Make a pull request
+
+## Code Style
+- We use [ESLint](https://eslint.org/) to enforce our code style; you should have this integrated into your editor / IDE
+- We also have enabled a pre-commit hook to run `eslint --fix --quiet` on all staged files. This will fix issues that can be fixed automatically and should prevent commits that violate the rules
+- We only have partial TypeScript adoption. It would be nice if new files were in TypeScript and those files were type safe, but it's not a requirement
+
+## Commit Style
+
+Please follow this guidance when committing to the main branch, including merge commits from PRs. Atomic commits in branches that don't make it into main can be more freeform.
+
+1. Try to use the [imperative mood](https://www.git-basics.com/docs/git-commit/commit-message-rules), so your message should fill in the blank in, "If applied, this commit will \_\_\_\_\_"
+1. Describe the impact on the end user if possible, so "Fix broken photo sharing in iOS 18" would be better than "Switch to a fork of photo-sharing-library-x"
+1. Append developer-relevant details on subsequent lines, e.g.
+    ```
+    Fix broken photo sharing in iOS 18
+
+    * Switch to fork of photo-sharing-library-x
+    * Minor UI touch-ups
+    ```
+1. Include `Closes #1234` on a subsequent line if your commit closes issue 1234; that will automatically close the issue on GitHub and helps link commits to the issues they address. Example
+    ```
+    Fix broken photo sharing in iOS 18
+
+    Closes #1234
+    ```
+
+1. Semantic commit labels are fine (e.g. `fix: broken photo sharing in iOS 18`), but we don't (yet) require them
 
 ## Submitting Changes
 1. Make sure your changes are up-to-date with the latest changes in the main repository.
-2. Make sure your code is well-tested and passes all unit and integration tests.
-3. It would be nice if you could run e2e tests locally with your own app credentials.
-4. Create a pull request to the main repository with your changes.
+1. Please make sure your code includes unit tests, and our linting rules were applied (should happen automatically on commit).
+1. Create a pull request to the main repository with your changes.
```

Comment:
should we say something about "you're responsible for making sure tests are passing in your PRs"? it seems like that's the biggest reason we don't end up merging some PRs in a timely manner

Reply:
I agree that we should include something about making sure tests are passing or if they aren't passing note why in case there is a blocker of some sort

Reply:
Will do.